sdks/python/apache_beam/io/bigquery.py

```diff
@@ -145,7 +146,14 @@ class RowAsDictJsonCoder(coders.Coder):
   """
 
   def encode(self, table_row):
-    return json.dumps(table_row)
+    # The normal error when dumping NAN/INF values is:
+    # ValueError: Out of range float values are not JSON compliant
+    # This code will catch this error to emit an error that explains
+    # to the programmer that they have used NAN/INF values.
+    try:
+      return json.dumps(table_row, allow_nan=False)
```

Comment:
I imagine whatever code they use to detect NAN and serialize it out will have to raise an exception instead. So I don't expect there to be anything noticeable.  Is there a benchmark or something you'd like me to run?